cobbler/actions/buildiso/__init__.py

```diff
@@ -172,30 +238,55 @@ def __copy_files(self, iso_distro, buildisodir: str = ""):
                 "Required file(s) not found. Please check your syslinux installation"
             )
 
-        self.logger.info("copying GRUB2 files")
-        bootloader_directory = pathlib.Path(self.api.settings().bootloaders_dir)
-        grub_efi = bootloader_directory.joinpath(
-            "grub", self.calculate_grub_name(iso_distro)
+    def _render_grub_entry(
+        self, append_line, menu_name, kernel_path, initrd_path
+    ) -> str:
+        return self.templar.render(
+            self.grub_menuentry_template,
+            out_path=None,
+            search_table={
+                "menu_name": menu_name,
+                "kernel_path": kernel_path,
+                "initrd_path": initrd_path,
+                "kernel_options": re.sub(r".*initrd=\S+", "", append_line),
+            },
         )
-        buildiso_directory = pathlib.Path(buildisodir)
-        grub_target = buildiso_directory.joinpath("grub", "grub.efi")
-        if grub_efi.exists():
-            filesystem_helpers.copyfile(str(grub_efi), str(grub_target), symlink=True)
+
+    def _render_isolinux_entry(
+        self, append_line: str, menu_name: str, kernel_path: str, menu_indent: int
+    ) -> str:
+        """Render a single isolinux.cfg menu entry."""
+        if menu_indent:
+            indent_statement = f"\n  MENU INDENT {menu_indent:d}"
         else:
-            self.logger.error('The following files were not found: "%s"', grub_efi)
-            raise FileNotFoundError(
-                "Required file(s) not found. Please check your GRUB 2 installation"
+            indent_statement = ""
+        return textwrap.dedent(
+            f"""
+            LABEL {menu_name}{indent_statement}
+              MENU LABEL {menu_name}
+              KERNEL {kernel_path}
+            {append_line}
+            """
+        )
```

Comment:
I would prefer to also have a template for that like GRUB has. Is there a reason not to have it in a template?

Reply:
Just laziness, I can migrate move it to a template.